spec/models/hunter_spec.rb

```diff
@@ -164,4 +164,22 @@
       it { is_expected.to be_falsey }
     end
   end
+
+  describe '#add_defaults' do
+    subject { hunter.moves.count }
```

Comment:
If your describe is `#add_defaults`, then the subject should be add_defaults. there are a couple of other improvements needed in this test.

This test and the test on line 176 are running on different sets of data. If the method you wrote appended all moves to the hunter, both tests would pass. If you create a basic move, and any other move at the top-level context, this becomes one sub-context, one with a hunter with a move, and the base one without a hunter move.

By the way, I'm not sure if you'll actually be able to create a hunter with a move before creating the hunter. Moves are associated with hunters through hunters_moves which have a hunter_id and move_id. Before it's created, the hunter does not have an id, so I'm not sure you'll be able to associate a move with a hunter before creation. There might be some Rails magic, but I don't know of it.

Your expectations also look for a count of records. It's better to check if `expect(hunter.moves).to include(moves_basic)` or something like that. This ensures that you've added the correct record to the Hunter.

Once we've modified `add_defaults` to add all `Moves::Basic` then this test can just add one instead of many moves_basic.

Reply:
Hopefully I modified the code as you expected, I was going for the test to cover two things
1. If the hunter was created with moves ( not sure if this would ever happen) it wouldn't modified the moves
2. If moves are empty it would add the Moves::Basic